cartographer/mapping/map_builder.cc

```diff
@@ -366,5 +367,18 @@ void MapBuilder::LoadState(io::ProtoStreamReaderInterface* const reader,
   CHECK(reader->eof());
 }
 
+void MapBuilder::LoadStateFromFile(const std::string& state_filename) {
+  // Check if suffix of the state file is ".pbstream".
+  const std::string suffix = ".pbstream";
+  CHECK_EQ(state_filename.substr(
```

Comment:
Looks a bit hacky, is there no more sophisticated way to check if the file format is correct?

Reply:
I just copied this function from cartographer_ros. On a second thought, it should only warn if the suffix differs because checks should only fail if the algorithm won't work. They should fail if the contents cannot be loaded, and this LoadState will do.